### PR TITLE
MI tests fixed

### PR DESCRIPTION
**Pull request by milanbrkic-ms**
## Description

Re-recorder MI tests that were depending on production resources.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@MilanBrkicFON  
Please check the following error
```
2021-01-29T09:55:05.2199648Z   [xUnit.net 00:00:45.29]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.VulnerabilityAssessmentMiTests.VulnerabilityAssessmentManagedDatabaseScanRecordGetListTest [FAIL]
2021-01-29T09:55:06.2360320Z     X Microsoft.Azure.Commands.Sql.Test.ScenarioTests.VulnerabilityAssessmentMiTests.VulnerabilityAssessmentManagedDatabaseScanRecordGetListTest [3s 363ms]
2021-01-29T09:55:06.2466193Z EXEC : error Message:  [D:\a\1\s\build.proj]
2021-01-29T09:55:06.2467638Z      System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Cannot create instance of management client type StorageManagementClient. It does not have the expected constructor.
2021-01-29T09:55:06.2468733Z     Stack Trace:
```

**Comment by MilanBrkicFON** [comment]
@msJinLei I'll record those tests again.

**Comment by msJinLei** [comment]
> @msJinLei I'll record those tests again.

It doesn't look like a record mismatch error. You may check the storage client creation in your test framework to see whether there is an incompatibility.

**Comment by msJinLei** [comment]
> @msJinLei I'll record those tests again.

One test case still fails in Linux. Please take a look.

```
2021-02-03T13:19:30.7547862Z   [xUnit.net 00:06:27.46]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ManagedDatabaseLogReplayScenarioTest.TestPipingCompleteCancelManagedDatabaseLogReplayService [FAIL]
2021-02-03T13:19:32.8005826Z     X Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ManagedDatabaseLogReplayScenarioTest.TestPipingCompleteCancelManagedDatabaseLogReplayService [2s 148ms]
2021-02-03T13:19:32.8080947Z EXEC : error Message:  [/home/vsts/work/1/s/build.proj]
2021-02-03T13:19:32.8083712Z      System.Management.Automation.RuntimeException : Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:32.8087672Z   ---- System.Management.Automation.RuntimeException : Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8010321Z     Stack Trace:
2021-02-03T13:19:33.8034610Z        at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2021-02-03T13:19:33.8035709Z      at System.Management.Automation.Runspaces.Pipeline.Invoke()
2021-02-03T13:19:33.8036660Z      at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2021-02-03T13:19:33.8038296Z      at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2021-02-03T13:19:33.8039312Z      at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-03T13:19:33.8040445Z      at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-03T13:19:33.8041877Z      at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-03T13:19:33.8042749Z      at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2021-02-03T13:19:33.8043469Z      at System.Management.Automation.PowerShell.Invoke()
2021-02-03T13:19:33.8044466Z      at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in /home/vsts/work/1/s/tools/ScenarioTest.ResourceManager/EnvironmentSetupHelper.cs:line 573
2021-02-03T13:19:33.8046040Z      at Microsoft.Azure.Commands.ScenarioTest.SqlTests.SqlTestsBase.RunPowerShellTest(String[] scripts) in /home/vsts/work/1/s/src/Sql/Sql.Test/ScenarioTests/SqlTestsBase.cs:line 108
2021-02-03T13:19:33.8047243Z      at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ManagedDatabaseLogReplayScenarioTest.TestPipingCompleteCancelManagedDatabaseLogReplayService() in /home/vsts/work/1/s/src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseLogReplayScenarioTest.cs:line 67
2021-02-03T13:19:33.8048837Z   ----- Inner Stack Trace -----
2021-02-03T13:19:33.8109974Z   
2021-02-03T13:19:33.8110743Z     Standard Output Messages:
2021-02-03T13:19:33.8111765Z    Test-PipingCompleteCancelManagedDatabaseLogReplay
2021-02-03T13:19:33.8113572Z    Caught Exception: System.Management.Automation.RuntimeException: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel ---> System.Management.Automation.RuntimeException: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8115529Z       --- End of inner exception stack trace ---
2021-02-03T13:19:33.8116465Z       at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2021-02-03T13:19:33.8117299Z       at System.Management.Automation.Runspaces.Pipeline.Invoke()
2021-02-03T13:19:33.8118448Z       at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2021-02-03T13:19:33.8119524Z       at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2021-02-03T13:19:33.8121965Z       at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-03T13:19:33.8123218Z       at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-03T13:19:33.8124172Z       at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-03T13:19:33.8125306Z       at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2021-02-03T13:19:33.8126077Z       at System.Management.Automation.PowerShell.Invoke()
2021-02-03T13:19:33.8126979Z       at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in /home/vsts/work/1/s/tools/ScenarioTest.ResourceManager/EnvironmentSetupHelper.cs:line 573
2021-02-03T13:19:33.8128105Z    Message: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8129459Z EXEC : PowerShell error Record: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel [/home/vsts/work/1/s/build.proj]
2021-02-03T13:19:33.8130865Z    Exception:System.Management.Automation.RuntimeException: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8131709Z    Details:
2021-02-03T13:19:33.8132772Z    Script Stack Trace: at Assert-Null, /home/vsts/work/1/s/src/Sql/Sql.Test/bin/Debug/netcoreapp2.1/Assert.ps1: line 350
2021-02-03T13:19:33.8134186Z    at Test-PipingCompleteCancelManagedDatabaseLogReplay, /home/vsts/work/1/s/src/Sql/Sql.Test/bin/Debug/netcoreapp2.1/ScenarioTests/ManagedDatabaseLogReplayScenarioTest.ps1: line 347
2021-02-03T13:19:33.8135388Z    at <ScriptBlock>, <No file>: line 1
2021-02-03T13:19:33.8136327Z    : Target: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8137099Z    
2021-02-03T13:19:33.8137963Z    Caught Exception: System.Management.Automation.RuntimeException: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8139109Z    Message: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8140213Z EXEC : PowerShell error Record: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel [/home/vsts/work/1/s/build.proj]
2021-02-03T13:19:33.8141446Z    Exception:System.Management.Automation.ParentContainsErrorRecordException: Assertion failed because the object is not null: Microsoft.Azure.Commands.Sql.ManagedDatabase.Model.AzureSqlManagedDatabaseModel
2021-02-03T13:19:33.8142354Z    Details:
2021-02-03T13:19:33.8143035Z    Script Stack Trace: 
2021-02-03T13:19:33.8143605Z    : Target: 
2021-02-03T13:19:33.8144210Z    
```

**Comment by MilanBrkicFON** [comment]
I am not sure how to fix Assert that is passing Windows tests and failing on Linux. Can you re-run pipeline or give me a hint? In local everything pass.

File: src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseLogReplayScenarioTest.ps1 [hunk 1]
```diff
@@ -380,8 +334,8 @@ function Test-PipingCompleteCancelManagedDatabaseLogReplay
 		
 		try {
 			$dbRemoved = Get-AzSqlInstanceDatabase `
-				-ResourceGroupName $rgName `
-				-InstanceName $managedInstance `
+				-ResourceGroupName $resourceGroupName `
+				-InstanceName $managedInstanceName `
```
Comment (msJinLei) [inline src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseLogReplayScenarioTest.ps1:338]: According to the error message, the test case fails in the line of "Assert-Null $dbRemoved“，where $dbRemoved is not null.

Comment (msJinLei) [reply src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseLogReplayScenarioTest.ps1:338]: Currently I can see from the error log is that the db that should be removed still can be retrieved by Get-AzSqlInstanceDatabaseLogReplay.
There are something can be done for the issue check.
- please make sure the test case does run with no error in windows. Sometime the exception is swallowed by catch.
- please also check whether the path join, if applicable, in the implementation of the cmdlets used in the test case also works well in Linux.

**Comment by msJinLei** [comment]
@MilanBrkicFON we are going to close the PR since no update for a long time. If you want to continue your work,  feel free to submit a new one.

**Pull request closed without merging**